openfeature/client.py

```diff
@@ -61,26 +60,29 @@
 
 @dataclass
 class ClientMetadata:
-    name: typing.Optional[str]
+    name: typing.Optional[str] = None
+    domain: typing.Optional[str] = None
 
 
 class OpenFeatureClient:
     def __init__(
         self,
-        name: typing.Optional[str],
+        domain: typing.Optional[str],
         version: typing.Optional[str],
-        provider: FeatureProvider,
         context: typing.Optional[EvaluationContext] = None,
         hooks: typing.Optional[typing.List[Hook]] = None,
     ) -> None:
-        self.name = name
+        self.domain = domain
         self.version = version
         self.context = context or EvaluationContext()
         self.hooks = hooks or []
-        self.provider = provider
+
+    @property
+    def provider(self) -> FeatureProvider:
+        return api.get_provider(domain=self.domain)
```

Comment:
you probably need to also implement `@provider.setter` otherwise you wouldn't be able to change it via the attribute anymore, except it is intended.